### PR TITLE
[Integrations] Jupiter sdk

### PR DESCRIPTION
**Pull request by 0xPratik**

### CODE REVIEW

**Comment by 0xPratik** [comment]
Hey @jacobcreech can you please review this PR

**Comment by 0xPratik** [comment]
Hey @jacobcreech could you please review this PR as well

File: code/jupiter/route-map/main.en.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  const tokens: Token[] = await (await fetch(TOKEN_LIST_URL[ENV])).json();
+
+  const USER_KEYPAIR = Keypair.generate();
+  const connection = new Connection("https://api.devnet.solana.com");
+
+  const jupiter = await Jupiter.load({
+    connection,
+    cluster: ENV,
+    user: USER_KEYPAIR, // or public key
+    // platformFeeAndAccounts:  NO_PLATFORM_FEE,
```
Comment (jacobcreech) [inline code/jupiter/route-map/main.en.ts:25]: Remove comments and only provide info once in the page please

File: code/jupiter/swap/main.preview.en.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+bestRoute = routes.routesInfos[0];
+const { execute } = await jupiter.exchange({
+  bestRoute,
+});
+
+// Execute Swap
+const swapResult: any = await execute();
```
Comment (jacobcreech) [inline code/jupiter/swap/main.preview.en.ts:7]: Why `any`?

File: code/jupiter/useJupiter/main.en.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  const jupiter = useJupiter({
+    amount: 1 * 10 ** 6, // raw input amount of tokens
+    inputMint,
+    outputMint,
+    slippage: 1, // 1% slippage
+    debounceTime: 250, // debounce ms time before refresh
+  });
+
+  const {
+    allTokenMints, // all the token mints that is possible to be input
```
Comment (jacobcreech) [inline code/jupiter/useJupiter/main.en.ts:28]: Please explain comments in the actual words above the code, not the code itself

File: docs/integrations/jupiter.md [hunk 4]
```diff
@@ -0,0 +1,438 @@
...
+      content: index,follow
+---
+
+# Jupiter
+
+Jupiter is the key liquidity aggregator for Solana, offering the widest range of tokens and best route discovery between any token pair.
+
+## How to use Jupiter Core Library
+
+Jupiter Core SDK is used to interact with the Jupiter on-chain programs, which gives us a list of best possible routes for performing a token swap.
```
Comment (jacobcreech) [inline docs/integrations/jupiter.md:45]: Is this needed? We could remove this part and leave the installation blurb

File: docs/integrations/jupiter.md [hunk 5]
```diff
@@ -0,0 +1,438 @@
...
+
+  </template>
+
+  </SolanaCodeGroupItem>
+
+</SolanaCodeGroup>
+
+### Loading the Jupiter instance
+
+Jupiter instace is being created with the provided configurations. There are alot of options present to tinker with to know more about it. go [here](https://docs.jup.ag/jupiter-core/full-guide)
```
Comment (jacobcreech) [inline docs/integrations/jupiter.md:94]: What is a jupiter instance?

File: docs/integrations/jupiter.md [hunk 6]
```diff
@@ -0,0 +1,438 @@
...
+```bash
+npm install @jup-ag/react-hook
+```
+
+  </CodeGroupItem>
+</CodeGroup>
+
+### Adding the Provider
+
+We need to setup the JupiterProvider in order to use the useJupiter Hook Through the React App.
```
Comment (jacobcreech) [inline docs/integrations/jupiter.md:204]: Why suddenly jumping into React?

File: docs/integrations/jupiter.md [hunk 7]
```diff
@@ -0,0 +1,438 @@
...
+
+  </template>
+
+  </SolanaCodeGroupItem>
+
+</SolanaCodeGroup>
+
+### Fetching the List of Tokens
+
+All the possible Tokens that can be swapped for a Given Network is fetched.
```
Comment (jacobcreech) [inline docs/integrations/jupiter.md:227]: Why is this done again?

**Review by jacobcreech — CHANGES_REQUESTED** [review]
Need more context on each code snippet. Probably remove the react specific parts

File: docs/integrations/jupiter.md (hunk 6 above)
Comment (dicksonpys) [reply docs/integrations/jupiter.md]: Hi, I'm Dickson from Jupiter,
Jupiter have different library that caters to the different needs of the client (web, mobile, native), therefore @0xPratik included different section/method of interacting with Jupiter.

There's mainly three way you can interact with Jupiter right now: 
1. Core
2. React hook
3. API 

Let us know if that clears up a little.

File: docs/integrations/jupiter.md (hunk 7 above)
Comment (dicksonpys) [reply docs/integrations/jupiter.md]: I believe this is intended as a helper section for developers wondering what/which pairs are possible on Jupiter.

Not any routes/mint are swappable on Jupiter as it has to reach certain liquidity threshold, therefore, we think having such a list might be very helpful for developers.

File: docs/integrations/jupiter.md (hunk 5 above)
Comment (dicksonpys) [reply docs/integrations/jupiter.md]: Jupiter instance are instantiated with some essentials parameters, Connection, cluster, and User's wallet.

1. We want to allow user to specify their own RPC, be it public or private RPC
2. To support different env
3. To generate/sign the required txs


Once instantiated, user can access the instance's underlying function which powers the swap.
1. Getting routeMap `jupiter.getRouteMap()`, to know which pairs are swappable on Jupiter
2. Computing route `jupiter.computeRoute()`, to get a list of routes/path that can be used
3. To Swap `jupiter.execute()`, performing swap

**Review by jacobcreech — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**